maven/lib/dependabot/maven/file_parser/property_value_finder.rb

```diff
@@ -69,6 +73,16 @@ def property_details(property_name:, callsite_pom:)
 
         attr_reader :dependency_files
 
+        def extract_value_from_expression(expression_value:, property_name:, callsite_pom:)
+          # and the expression is pointing to self then raise the error
+          if expression_value.eql?("${#{property_name}}")
+            raise Dependabot::DependencyFileNotParseable, "[ERROR] Resolving expression: '#{expression_value}'"
+          end
+
+          # and the expression is pointing to another tag, then get the value of that tag
+          property_details(property_name: expression_value.slice!(2..-2), callsite_pom: callsite_pom)
```

Comment:
```suggestion
          property_details(property_name: expression_value.slice(2..-2), callsite_pom: callsite_pom)
```

`slice!` with a `!` will mutate the value in place, there is no need to mutate the value in place so we can use the non-mutating form. While in this case it won't change the behavior of the code, in general it's considered good practice to use non-mutating forms when possible, as mutating data can be a source of bugs.